sh/drive-utils.sh

```diff
@@ -0,0 +1,526 @@
+    fi
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    { folder_id_create_directory="${folder_id_create_directory:-$(printf "%s\n" "${create_folder_response_create_directory}" | _json_value id 1 1)}" && printf "%s\n" "${folder_id_create_directory}"; } ||
+        { printf "%s\n" "${create_folder_response_create_directory}" 1>&2 && return 1; }
+    return 0
+)
+
+###################################################
+# Sub functions for _upload_file function - Start
+# generate resumable upload link
+_generate_upload_link() {
+    "${EXTRA_LOG:-:}" "justify" "Generating upload link.." "-" 1>&2
+    uploadlink_upload_file="$(curl --compressed "${CURL_PROGRESS_EXTRA}" \
+        -X "${request_method_upload_file}" \
+        -H "Authorization: Bearer ${token_upload_file}" \
+        -H "Content-Type: application/json; charset=UTF-8" \
+        -H "X-Upload-Content-Type: ${mime_type_upload_file}" \
+        -H "X-Upload-Content-Length: ${inputsize_upload_file}" \
+        -d "$postdata_upload_file" \
+        "${url_upload_file}" \
+        -D - || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    uploadlink_upload_file="$(printf "%s\n" "${uploadlink_upload_file}" | sed -n 's/[L,l]ocation: //p' | sed "s/$(printf '\r')//g")"
+    { [ -n "${uploadlink_upload_file}" ] && return 0; } || return 1
+}
+
+# Curl command to push the file to google drive.
+_upload_file_from_uri() {
+    _print_center "justify" "Uploading.." "-"
+    # shellcheck disable=SC2086 # Because unnecessary to another check because ${CURL_PROGRESS} won't be anything problematic.
+    upload_body_upload_file="$(curl --compressed ${CURL_PROGRESS} \
+        -X PUT \
+        -H "Authorization: Bearer ${token_upload_file}" \
+        -H "Content-Type: ${mime_type_upload_file}" \
+        -H "Content-Length: ${content_length_upload_file}" \
+        -H "Slug: ${slug_upload_file}" \
+        -T "${input_upload_file}" \
+        -o- \
+        --url "${uploadlink_upload_file}" \
+        --globoff \
+        ${CURL_SPEED} ${resume_args_upload_file} || :)"
+    [ -z "${VERBOSE_PROGRESS}" ] && for _ in 1 2; do _clear_line 1; done
+    return 0
+}
+
+# logging in case of successful upload
+_normal_logging_upload() {
+    [ -z "${VERBOSE_PROGRESS}" ] && _clear_line 1
+    "${QUIET:-_print_center}" "justify" "${slug_upload_file} " "| ${readable_size_upload_file} | ${STRING}" "="
+    return 0
+}
+
+# Tempfile Used for resuming interrupted uploads
+_log_upload_session() {
+    [ "${inputsize_upload_file}" -gt 1000000 ] && printf "%s\n" "${uploadlink_upload_file}" >| "${__file_upload_file}"
+    return 0
+}
+
+# remove upload session
+_remove_upload_session() {
+    rm -f "${__file_upload_file}"
+    return 0
+}
+
+# wrapper to fully upload a file from scratch
+_full_upload() {
+    _generate_upload_link || { _error_logging_upload "${slug_upload_file}" "${uploadlink}" && return 1; }
```

Comment:
Codacy found an issue: [uploadlink is referenced but not assigned.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)